backend/grants/templates/admin/grants/grant_summary.html

```diff
@@ -0,0 +1,163 @@
+    <table id="grantsByCountry">
+        <thead>
+            <tr>
+                <th>Continent</th>
+                <th>Country</th>
+                {% for status in statuses %}<th>{{ status.1|title }}</th>{% endfor %}
+            </tr>
+        </thead>
+        <tbody>
+            <tbody>
+                {% for key, counts in summary.items %}
+                    {% with continent=key.0 country=key.1 %}
+                        {% ifchanged continent %}
+                            <tr class="continent-total">
+                                <td colspan="2">
+                                    <strong>Total for {{ continent }}:</strong>
+                                </td>
+                                {% for status in statuses %}
+                                    <td>
+                                        <strong>{{ totals_per_continent|get_item:continent|get_item:status.0|default:"0" }}</strong>
+                                    </td>
+                                {% endfor %}
+                            </tr>
+                        {% endifchanged %}
+                        <tr>
+                            <td>{{ continent }}</td>
+                            <td>{{ country }}</td>
+                            {% for status in statuses %}<td>{{ counts|get_item:status.0|default:"0" }}</td>{% endfor %}
+                        </tr>
+                    {% endwith %}
+                {% endfor %}
+            </tbody>
+            <tfoot>
+                <tr>
+                    <td colspan="2">Total Grants: {{ total }}</td>
+                    {% for count in totals_by_status.values %}<td>{{ count }}</td>{% endfor %}
+                </tr>
+            </tfoot>
+        </table>
+        <h2>by Gender</h2>
+        <table id="grantsByGender">
+            <thead>
+                <tr>
+                    <th>Gender</th>
+                    <th></th>
+                    {% for status in statuses %}<th>{{ status.1|title }}</th>{% endfor %}
+                </tr>
+            </thead>
+            <tbody>
+                {% for gender, gender_data in gender_summary.items %}
+                    <tr>
+                        <td>{{ genders|get_item:gender|default:"Not Specified" }}</td>
+                        <td></td>
+                        {% for status in statuses %}<td>{{ gender_data|get_item:status.0|default:"0" }}</td>{% endfor %}
+                    </tr>
+                {% endfor %}
+            </tbody>
+        </table>
+        <script>
+        document.addEventListener('DOMContentLoaded', function() {
+            var checkboxes = document.querySelectorAll('.toggle-column');
+            checkboxes.forEach(function(checkbox) {
+                checkbox.addEventListener('change', function() {
+                    var countryColumn = document.getElementById('grantsByCountry').querySelectorAll('td:nth-child(' + checkbox.dataset.column + '),th:nth-child(' + checkbox.dataset.column + ')');
```

Comment:
why are you using both `getElementById` and `querySelectorAll`? can you not just use `querySelectorAll`?

Reply:
or maybe using JS template strings would look better